src/azure/azure-mesh-adapter.ts

```diff
@@ -80,27 +80,34 @@ export class AzureMeshAdapter implements MeshAdapter {
       costInformations.push(...costInformation);
     }
 
-    let currencySymbol: string | null = null;
     const summedCosts = new Map<string, number>();
+    const currencySymbols = new Map<string, string>();
     for (const ci of costInformations) {
-      // TODO manage the currency symbol.
-      if (currencySymbol === null || currencySymbol === ci.currency) {
-        currencySymbol = ci.currency;
+      if (!currencySymbols.has(ci.subscriptionId)) {
+        currencySymbols.set(ci.subscriptionId, ci.currency);
       } else {
-        throw new MeshAzurePlatformError(
-          AzureErrorCode.AZURE_CLI_GENERAL,
-          "Encoutered two different currency during cost collection. This is currently not supported.",
-        );
+        // Make sure we only collect the same currency for one tenant.
+        // Multiple currency for the same tenant are currently not supported.
+        if (currencySymbols.get(ci.subscriptionId) !== ci.currency) {
+          throw new MeshAzurePlatformError(
+            AzureErrorCode.AZURE_CLI_GENERAL,
+            "Encoutered two different currency during cost collection. This is currently not supported.",
```

Comment:
```suggestion
            "Encountered two different currencies within one Subscription during cost collection. This is currently not supported.",
```

Reply:
Right? That's the issue (which we do not expect)

Reply:
yep